### PR TITLE
[master < T1209] Fix API bug on accessing deleted object

### PR DESCRIPTION
**Pull request by antoniofilipovic**
Issue: When doing detach delete on graph, and accessing deleted objects from C++ API, if you are accessing `From` and `To` vertex of deleted edge, error occurs that you are accessing deleted vertex although it shouldn't.

When a vertex is detach deleted, its VertexAccessor is marked `for_deleted_`, and  out|in EdgeAccessors are marked `for_deleted_`. VertexAccessors which you can access from EdgeAccessor are not marked for_deleted_.  This creates a slight problem when working with deleted objects in API, because only EdgeAccessor is marked `for_deleted_` not also its `VertexAccessors`.  Therefore some `VertexAccessors` have wrong information, marking `for_deleted_` as false instead of true.

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [ ] Write a release note here, including added/changed clauses
- [ ] Tag someone from docs team in the comments

### CODE REVIEW

**Review by antepusic — COMMENTED** [review]
Everything looks fine, but I’d prefer to overload `FromVertex` and `ToVertex` instead of creating new methods. The new ones’ names (`FromVertexWithDeleted` and `ToVertexWithDeleted`) really aren’t very clear, and tbh I can’t think of a name that’s both clear and descriptive.

File: src/query/db_accessor.hpp [hunk 1]
```diff
@@ -94,6 +94,10 @@ class EdgeAccessor final {
 
   VertexAccessor From() const;
 
+  VertexAccessor ToWithDeleted() const;
```
Comment (Josipmrden) [inline src/query/db_accessor.hpp:97]: What does this mean

File: src/storage/v2/edge_accessor.cpp [hunk 2]
```diff
@@ -105,6 +105,15 @@ VertexAccessor EdgeAccessor::ToVertex() const {
   return VertexAccessor{to_vertex_, transaction_, indices_, constraints_, config_};
 }
 
+VertexAccessor EdgeAccessor::FromVertexWithDeleted() const {
+  return VertexAccessor{from_vertex_, transaction_, indices_,
+                        constraints_, config_,      for_deleted_ && from_vertex_->deleted};
```
Comment (Josipmrden) [inline src/storage/v2/edge_accessor.cpp:110]: I'm seeing a lot of spaces here

File: src/query/db_accessor.hpp (hunk 1 above)
Comment (Josipmrden) [reply src/query/db_accessor.hpp:97]: I'd like this to be documented

**Review by Josipmrden — APPROVED** [review]

**Comment by vpavicic** [comment]
@antoniofilipovic - can you write me a release note (RN)?

**Comment by antoniofilipovic** [comment]
This PR enables to access deleted vertex from deleted edge when being processed as part of trigger function

**Pull request merged**